### ISSUE TITLE
Tales of Zestiria the X [tvdb5-311014]

### ISSUE THREAD

**Issue by DJScias** [ZeroQI/Hama.bundle#373]
Hello there,

I was looking around on TVBD and noticed it's using an "Official Order" (https://www.thetvdb.com/series/tales-of-zestiria-the-x/allseasons/official) where Special 2 is episode 0.

I tried to emulate this behaviour, sort of, with tvdb5.. As I have no clue what other method I should use to emulate this.

This however ended up with the following:
![image](https://user-images.githubusercontent.com/439655/69253740-66ce3e80-0bb5-11ea-951b-53e0a00222d9.png)

![image](https://user-images.githubusercontent.com/439655/69253679-4e5e2400-0bb5-11ea-99ac-3ae4228f682a.png)
![image](https://user-images.githubusercontent.com/439655/69253689-51591480-0bb5-11ea-9957-ef9ffc6000d2.png)

So while it is attaching the right numbers and names (I guess it shifted episode 1 to 2 in Plex) it's keeping the special in the specials season and that's sort of defeating the purpose.

Should I use another tvdbX method? Or what should I do here?

As probably required, the attached logs:
[Tales of Zestiria the X [tvdb5-311014].agent-update.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/3870052/Tales.of.Zestiria.the.X.tvdb5-311014.agent-update.log)
[Tales of Zestiria the X [tvdb5-311014].agent-search.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/3870053/Tales.of.Zestiria.the.X.tvdb5-311014.agent-search.log)

Thanks in advance.

**Comment by ZeroQI** [comment]
It seems that when i changed from absolute numbering to tvdb absolute index, didn't update the scanner and agent enough...
- Scanner issue fixed.
- Transfered to HAMA for software issue

**Comment by DJScias** [comment]
Alright, do let me know if and when I gotta test something anew.

**Comment by DJScias** [comment]
Just checking in, has any progress been made on this?

**Comment by ZeroQI** [comment]
Not yet. Away from PC while away.

**Comment by DJScias** [comment]
Checking in once more, hope you're doing well ZeroQI.

Is there anything I need to provide or the like that could help you guys out?

**Comment by ZeroQI** [comment]
Summary of previous changes
- Scanner fix was tvdb5 mappingList missing specials so obviously not remapped, occuring Line 656:
from if episode.xpath('SeasonNumber')[0].text != '0' and episode.xpath('absolute_number')[0].text:
to if episode.xpath(episode.xpath('absolute_number')[0].text:
- https://github.com/ZeroQI/Absolute-Series-Scanner/blob/e180c1f1d5f87c26332d9540d56a8bf2362f4a13/Scanners/Series/Absolute%20Series%20Scanner.py line 656

Now for Hama changes, looking at Hama code, the magic happens https://github.com/ZeroQI/Hama.bundle/blob/master/Contents/Code/TheTVDBv2.py from line 174

Please change line 175 from
```
        episode = str(abs_number)
```
to
```
        if abs_number:  season, episode = '1', str(abs_number)
```
In other words, if there is an absolute number for the episode in thetvdb it is s01exx otherwise it's a special nothing changes. The way it was, a special s00e01 with an absolute index of 10 would end up s00e10 which is wrong...

Please test and report the above fix
Please try and report

**Comment by DJScias** [comment]
I triggered a `Refresh Metadata` on the series on Plex and it all sorted itself.

![image](https://user-images.githubusercontent.com/439655/77225486-deb27280-6b6f-11ea-97af-53f49283b42b.png)

Thanks again for fixing the issue.
